smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigProperty.kt

```diff
@@ -180,6 +180,11 @@ sealed class ClientConfigPropertyType {
      * @param default the value to assign if the corresponding builder property is null
      */
     data class RequiredWithDefault(val default: String) : ClientConfigPropertyType()
+
+    /**
+     * A configuration property that uses [render] to manually render the actual immutable property
+     */
+    data class Custom(val render: (ClientConfigProperty, KotlinWriter) -> Unit) : ClientConfigPropertyType()
```

Comment:
**Question**: Does anything about this custom type _require_ the property be immutable as stated in the KDoc?

Reply:
Technically no (although all property types are currently rendered as `val`). 

The alternative here to `Custom` would have been to allow existing property types to have an optional `mapping` argument that apples a mapping to the builder.